### DISCUSSION TITLE
GPT-3 Use Case Ideas

### DISCUSSION THREAD

**Discussion by trieloff** [adobe/helix-home#232] [Ideas]
I have gained access to a beta API key for GPT-3, OpenAI's version three of a pre-trained model for text generation, classifications, and search. Here are a few ideas and use cases I can see:

1. Metadata completion: authors create an article, and leave some metadata properties empty (or filled in with `...`). Upon pressing a "complete" shortcut or button we then send the text to the API, with a matching prompt.

Taking the plain text from https://blog.adobe.com/en/publish/2021/11/08/tech-empowers-fashion.plain.html and using following prompts (in bold) we get these results:

* **Target Audience for the Article:** Designers, Fashionistas
* **Hashtag:** #AdobeFashionWeek
* **tl;dr:** Fashion is a language of creativity and the industry has always had a deep relationship with technology
* **What this means for Adobe:** Adobe Creative Cloud is the go-to platform for fashion designers and brands
* **Promote the article with the following tweet:** How tech is empowering fashion https://www.adobe.com/us/marketing-cloud/blog… via @AdobeLive #COVID19 (note: the URL is guessed by the AI and could be replaced in post-processing)
* **Key quote from the article:** “Technology is a paper for me to write on. It's a tool. And as long as it is a tool that I can write my story with, I'm always excited to explore it." — Iris van Herpen
* **People mentioned in the article:** Karl Lagerfeld, Christina Najjar (aka Tinx) 
* **Stock image of** a model wearing a dress designed by Iris van Herpen in Adobe Creative Cloud. (feed this into Adobe Stock search)

2. Helix Bot knowledge base: Feed the searches API (https://beta.openai.com/docs/api-reference/searches) with the contents of the "howto" section of the Helix Website, then rely questions that contain the phrase "how to _x_ with Helix" or "how do I _x_ with Helix". Reply with a link as `@HelixBot`, or escalate to the team if no canonical answer exists

3. Helix bot block PRs: when a developer creates a `blocks/spinner/spinner.js` file in GitHub, create a PR for a Codex API completion (same model used for GitHub copilot) that bootstraps the implementation. **Note:** there are probably too few examples for the AI to create meaningful code here. There are slightly better results when framing the problem as a Web Components custom element or a React component, but providing the example HTML and expecting the AI to understand it in a meaningful way may be too much to ask for.

Other things I've tried: 
- fill out features, benefits, example customers and a testimonial quote for the Helix homepage. Believable if, but too generic.
- expand the list of tutorials on https://pages.adobe.com/illustrator/en/tl/thr-illustration-home/ This works great, but would need authors to fill out the rest. Maybe it can be combined with RUM data to identify content that is in demand, but not there yet
- generate movie synopsis and cast list from altered movie titles, years, and directors. Useless, but fun
